kubelet config controller isn't failing correctly

While working on #611 I got unit failures:

```
--- FAIL: TestContainerRuntimeConfigUpdate (3.73s)
    --- FAIL: TestContainerRuntimeConfigUpdate/aws (1.26s)
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc0003a0640}]
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc0001b5680}]
    --- FAIL: TestContainerRuntimeConfigUpdate/none (1.25s)
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc00012c280}]
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc00012c000}]
    --- FAIL: TestContainerRuntimeConfigUpdate/unrecognized (1.23s)
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc0001b5e00}]
        container_runtime_config_controller_test.go:228: error syncing containerruntimeconfigs: PatchType is not supported
        container_runtime_config_controller_test.go:73: 1 unexpected actions: [{ActionImpl:{Namespace: Verb:update Resource:{Group:machineconfiguration.openshift.io Version:v1 Resource:containerruntimeconfigs} Subresource:status} Object:0xc0001b5e00}]
```

That failure is totally legit but it's super weird that we're not getting the same error in kubelet_config which is doing almost the same as the container_runtime controller.

Turns out, when something fails in the kubelet_config controller, we call `syncStatusOnly`.

Now, in the container_runtime controller, on error, we call that and correctly bubble up the error. Whether for the kubelet_config, on error, we call UpdateStatus, that call goes well and we return `nil`. That's why I'm not getting the same error in units.

--- a/pkg/controller/kubelet-config/kubelet_config_controller.go
+++ b/pkg/controller/kubelet-config/kubelet_config_controller.go
@@ -333,7 +333,7 @@ func (ctrl *Controller) generateOriginalKubeletConfig(role string) (*ignv2_2type
 }
 
 func (ctrl *Controller) syncStatusOnly(cfg *mcfgv1.KubeletConfig, err error, args ...interface{}) error {
-	return retry.RetryOnConflict(updateBackoff, func() error {
+	statusUpdateError := retry.RetryOnConflict(updateBackoff, func() error {
 		newcfg, getErr := ctrl.mckLister.Get(cfg.Name)
 		if getErr != nil {
 			return getErr
@@ -342,6 +342,10 @@ func (ctrl *Controller) syncStatusOnly(cfg *mcfgv1.KubeletConfig, err error, arg
 		_, lerr := ctrl.client.MachineconfigurationV1().KubeletConfigs().UpdateStatus(newcfg)
 		return lerr
 	})
+	if statusUpdateError != nil {
+		glog.Warningf("error updating kubeletconfig status: %v", statusUpdateError)
+	}
+	return err
 }
 
 // syncKubeletConfig will sync the kubeletconfig with the given key.

